### PR TITLE
update gitlab label's color

### LINKED ISSUE #8991: community.general.gitlab_label does not update colors for existing labels
### Summary

When I try to update an existing gitlab label the change of a color is detected but not applied

### Issue Type

Bug Report

### Component Name

community.general.gitlab_label

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.17.2]
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
community.general 9.4.0
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
CONFIG_FILE() = None
PAGER(env: PAGER) = less
```

### OS / Environment

Ubuntu 22 in WSL

### Steps to Reproduce

```yaml (paste below)
- name: Set or update some labels
  community.general.gitlab_label:
    api_url: https://gitlab.com
    api_token: secret_access_token
    project: "group1/project1"
    labels:
      - name: label_one
        color: "#224488"
    state: present
```

### Expected Results

if the label already exist it should have the color 224488

### Actual Results

```console (paste below)
label color does not change
```

assuming this is due to `color` is missing in the `update_label` method found at https://github.com/ansible-collections/community.general/blob/main/plugins/modules/gitlab_label.py#L266

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by gpongelli**
##### SUMMARY
Set Gitlab label's color when updating a label.
Fail if both `new_name` and `color` are missing when updating a label, one of them is required as described into [Gitlab API docs](https://docs.gitlab.com/ee/api/labels.html).

Fixes #8991 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
Gitlab Label

### CODE REVIEW

File: plugins/modules/gitlab_label.py [hunk 1]
```diff
@@ -268,13 +268,18 @@ def update_label(self, var_obj):
             return True, True
         _label = self.gitlab_object.labels.get(var_obj.get('name'))
 
+        if (var_obj.get('new_name') is None) and (var_obj.get('color') is None):
+            self._module.fail_json(msg="Failed to update label, new_name or color must be present: %s " % _label)
```
Comment (russoz) [inline plugins/modules/gitlab_label.py:272]: This might be better suited as a `required_one_of` clause for the `labels` parameter.

Adding something like:
```
required_one_of=[('new_name', 'color')]
```
after line 425 below. Beware that this might have other impacts, so take this with a grain of salt.

**Review by russoz — COMMENTED** [review]
Hi @gpongelli 
Thank you for the contribution! Please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

Also a possible improvement, but don't take my word on it, please do check.

Cheers

**Review by russoz — COMMENTED** [review]
Hi @gpongelli 
Thank you for the contribution! Please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

Also a possible improvement, but don't take my word on it, please do check.

Cheers

File: plugins/modules/gitlab_label.py (hunk 1 above)
Comment (gpongelli) [reply plugins/modules/gitlab_label.py:272]: Hi @russoz , I know about `required_one_of`, I tought about it, but it's not the case to use it because those field are not always required:
- new_name is a field useful only for label's editing
- color is requested in more payload, but not for all

**Comment by gpongelli** [comment]
> Hi @gpongelli Thank you for the contribution! Please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).


Added.

File: plugins/modules/gitlab_label.py (hunk 1 above)
Comment (russoz) [reply plugins/modules/gitlab_label.py:272]: Perfect, thanks for confirming.

File: changelogs/fragments/9010-edit-gitlab-label-color.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - Gilab label module - be able to update label's color (https://github.com/ansible-collections/community.general/pull/9010).
```
Comment (russoz) [inline changelogs/fragments/9010-edit-gitlab-label-color.yaml:2]:
```suggestion
  - gitlab_label - add feature to update label's color (https://github.com/ansible-collections/community.general/pull/9010).
```

**Comment by russoz** [comment]
Other than the adjustment in the changelog frag, LGTM

**Review by russoz — APPROVED** [review]
LGTM

File: plugins/modules/gitlab_label.py (hunk 1 above)
Comment (felixfontein) [inline plugins/modules/gitlab_label.py:272]: Why is this needed? This is a new requirement that wasn't there in the past (and thus a breaking change).

Comment (gpongelli) [reply plugins/modules/gitlab_label.py:272]: Hi @felixfontein , looking at [Gitlab API docs](https://docs.gitlab.com/ee/api/labels.html) seems it was there also [in the past (this is from gitlab 15.6) ](https://archives.docs.gitlab.com/15.6/ee/api/labels.html#edit-an-existing-label) but the python library used by this module does not mention [this constraint](https://python-gitlab.readthedocs.io/en/stable/gl_objects/labels.html):

> Update a label for a project:
> 
> ```
> # change the name of the label:
> label.new_name = 'bar'
> label.save()
> # change its color:
> label.color = '#112233'
> label.save()
> ```


I've done a fast check with following code:
```python
_project = gl.projects.get(...)
_l = _project.labels.get('my-label')
_l.description = "new descript"
_l.save()
```
it works, without any "color" or "name", so I'm going to remove this check.

Comment (felixfontein) [reply plugins/modules/gitlab_label.py]: My guess is that the documentation has been wrong for a longer time, and "At least one parameter is required, to update the label." is the only relevant part. I'd guess that the "yes if xxx is not provided" has been there before `description` and `priority` was added.

And in fact: https://gitlab.com/gitlab-org/gitlab-foss/-/commit/c8be7f1cf027d27bba50f6fa4fdeaee33e3f531f#fd69b4be4cca7d3b5760b77f0fc25ea1c57685ae_115_126

If you look at the code change in the same commit (further below), you can see that the original condition for the error message `'Required parameters "new_name" or "color" missing'` changed to also include `description`, without the error message being updated (line ~80 of lib/api/labels.rb). I guess this was later updated again by adding `priority`, without again updating the (already wrong) documentation of `name` and `color`.

File: changelogs/fragments/9010-edit-gitlab-label-color.yaml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - gitlab_label - add feature to update label's color (https://github.com/ansible-collections/community.general/pull/9010).
```
Comment (felixfontein) [inline changelogs/fragments/9010-edit-gitlab-label-color.yaml:2]: So is this a feature (as said here) or a bugfix (as said in the PR description)? I would say this is a bugfix, since changing the color was likely intended when adding support for color, but just not implemented.

Comment (gpongelli) [reply changelogs/fragments/9010-edit-gitlab-label-color.yaml:2]: Agree with bugfix .

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this in a few days.

**Review by markuman — APPROVED** [review]

**Comment by felixfontein** [comment]
@gpongelli thanks for your contribution!
@russoz @markuman thanks for reviewing!
